src/main/scala/com/sky/kafka/configurator/ConfigParsing.scala

```diff
@@ -0,0 +1,40 @@
+package com.sky.kafka.configurator
+
+import java.io.File
+
+import com.sky.kafka.configurator.error.InvalidArgsException
+import org.apache.kafka.clients.admin.AdminClientConfig
+import scopt.OptionParser
+
+import scala.util.Try
+import scala.collection.JavaConverters._
+
+object ConfigParsing {
+
+  private val parser = new OptionParser[AppConfig]("kafka-configurator") {
+    opt[File]('f', "file").required().valueName("<file>")
+      .action((x, c) => c.copy(file = x))
+      .text("Topic configuration file")
+      .validate(file => if (file.exists()) success else failure(s"$file does not exist."))
+
+    opt[String]("bootstrap-servers").required()
+      .action((x, c) => c.copy(bootstrapServers = x))
+      .text("Kafka brokers URLs for bootstrap (comma-separated)")
+
+    opt[Map[String, String]]("properties").optional()
+      .action((x, c) => c.copy(props = x))
+      .text("Kafka admin client config as comma-separated pairs")
+  }
+
+  def parse(args: Seq[String], envVars: Map[String, String]): Try[AppConfig] =
+    parser.parse(args, AppConfig()).map { appConfig =>
+      val providedConfig = appConfig.props ++ collectKafkaConfigFrom(envVars)
+      appConfig.copy(props = providedConfig.filterKeys(AdminClientConfig.configNames().asScala.contains))
+    }.toRight(InvalidArgsException).toTry
+
+  private def collectKafkaConfigFrom(envVars: Map[String, String]): Map[String, String] =
+    envVars.collect {
+      case (k, v) =>
+        k.split("KAFKA_").tail.mkString.replaceAll("_", ".").toLowerCase -> v
```

Comment:
when there is no "KAFKA_" element - you get ("", v) as a result,
when you get "ABCKAFKA_XYZ" env variable you treat it as  "KAFKA_XYZ" variable
is it intended? (probably it's no harm, just asking)

Reply:
i think we can live with that

Reply:
ok